packages/block-editor/src/components/inserter-draggable-blocks/index.js

```diff
@@ -52,6 +35,13 @@ const InserterDraggableBlocks = ( {
 		useDispatch( blockEditorStore )
 	);
 
+	const patternBlock = useMemo( () => {
+		return pattern?.type === INSERTER_PATTERN_TYPES.user &&
+			pattern?.syncStatus !== 'unsynced'
+			? [ createBlock( 'core/block', { ref: pattern.id } ) ]
+			: undefined;
+	}, [ pattern ] );
```

Comment:
Won't this always run, because `pattern` is a new object each render? Would it be worth checking equality of its properties instead?

Reply:
Good point, this can be improved. Fixed!